spec/index.bs

```diff
@@ -2821,6 +2822,26 @@ GPUDevice includes GPUObjectBase;
     : <dfn>queue</dfn>
     ::
         The primary {{GPUQueue}} for this device.
+
+    : <dfn>adapterInfo</dfn>
+    ::
+        Information about the physical adapter which created the [=device=] that this {{GPUDevice}} refers to.
+
+        For a given {{GPUDevice}}, the {{GPUAdapterInfo}} values exposed are constant over time.
+
+        The same object is returned each time. To create that object for the first time:
```

Comment:
We need to decide if this returns same object as the adapter (i.e. if `adapter.info === device.adapterInfo`). I think it should but @toji WDYT?

In order to do this we can just remove these steps, and `Set |gpuDevice|.{{GPUDevice/adapterInfo}} to |this|.{{GPUAdapter/info}}` in the last steps of `requestDevice()`.

Reply:
If we require the same object to be returned each time, then we also require the information to be persisted which seems unnecessary.

Do frameworks require the same object or are they just interested in the string?

Reply:
> Would propose we remove the constraint that the same object is returned each time, which seems not needed, at least initially.

I think using [`[SameObject]`](https://webidl.spec.whatwg.org/#SameObject) in this way is common practice (we already have it on `adapter.info`. It allows accessing `.info.x` `.info.y` etc without creating a whole new `GPUAdapterInfo` allocation each time (it looks like a lightweight operation and it should be). We also shouldn't plan to change this later as it could be technically breaking (even if unlikely - like you say applications should generally be interested only in the string, but they could attach expando properties here.).

Returning the same object from both `adapter.info` and `device.adapterInfo` is less important, but it's best for compatibility if we decide now.

Reply:
The use of `[SameObject]` here is fine IMO, especially given that we use it for quite a few other attributes.

As for using the same JS object for both the adapter and device, I don't know that I see much benefit? It does strike me as a little strange that I can modify an attribute of one object and have that change be reflected on an attribute of another object. Plus, I don't really see what benefit that requirement would have? The only thing I could imagine is that it would allow you to detect if a given device was created by a given adapter, which isn't really useful? (Unless you want to persist the same adapter info across multiple `requestAdapter()` results to indicate that it's the same adapter each time, which sounds like a bad idea.)

I think for the sake of simplicity it's likely best if the adapter and device have separate JS objects for their adapter into.

Reply:
OK, then let's keep `[SameObject]` and keep the two copies separate? -- No change needed.

WDYT? @mwyrzykowski

Reply:
Sounds great, sorry meant to remove my 'request changes' after the F2F. The discussion regarding [SameObject] was not a blocker for me